src/main/java/dk/alexandra/fresco/lib/statistics/DEASolver.java

```diff
@@ -0,0 +1,246 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * FRESCO uses SCAPI - http://crypto.biu.ac.il/SCAPI, Crypto++, Miracl, NTL,
+ * and Bouncy Castle. Please see these projects for any further licensing issues.
+ *******************************************************************************/
+package dk.alexandra.fresco.lib.statistics;
+
+import java.util.Arrays;
+import java.util.List;
+
+import dk.alexandra.fresco.framework.Application;
+import dk.alexandra.fresco.framework.MPCException;
+import dk.alexandra.fresco.framework.ProtocolFactory;
+import dk.alexandra.fresco.framework.ProtocolProducer;
+import dk.alexandra.fresco.framework.value.OInt;
+import dk.alexandra.fresco.framework.value.SInt;
+import dk.alexandra.fresco.lib.field.integer.BasicNumericFactory;
+import dk.alexandra.fresco.lib.field.integer.RandomFieldElementFactory;
+import dk.alexandra.fresco.lib.helper.ParallelProtocolProducer;
+import dk.alexandra.fresco.lib.helper.sequential.SequentialProtocolProducer;
+import dk.alexandra.fresco.lib.lp.LPFactory;
+import dk.alexandra.fresco.lib.lp.LPFactoryImpl;
+import dk.alexandra.fresco.lib.lp.LPPrefix;
+import dk.alexandra.fresco.lib.lp.LPTableau;
+import dk.alexandra.fresco.lib.lp.Matrix;
+import dk.alexandra.fresco.lib.math.integer.NumericBitFactory;
+import dk.alexandra.fresco.lib.math.integer.exp.ExpFromOIntFactory;
+import dk.alexandra.fresco.lib.math.integer.exp.PreprocessedExpPipeFactory;
+import dk.alexandra.fresco.lib.math.integer.inv.LocalInversionFactory;
+
+/**
+ * Protocol for solving DEA problems.
+ * 
+ * Given a dataset (two matrices of inputs and outputs)
+ * and a number of query vectors, the protocol
+ * will compute how well the query vectors perform compared
+ * to the dataset. 
+ * 
+ * The result/score of the computation must be converted to
+ * a double using Gauss reduction to be meaningful. See
+ * the DEASolverTests for an example.  
+ *
+ */
+public class DEASolver implements Application{
+
+	private static final long serialVersionUID = 7679664125131997196L;
+	private List<List<SInt>> targetInputs, targetOutputs;  
+	private List<List<SInt>> inputDataSet, outputDataSet; 
+	
+	private SInt[] optimal;
+	private OInt[] output;
+	
+	public DEASolver(
```

Comment:
Would love a bit more context/documentation here. What are the different things, and what should these be. 
But cool application :)